lib/ansible/modules/cloud/amazon/aws_direct_connect_gateway.py

```diff
@@ -0,0 +1,375 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+module: aws_direct_connect_gateway
+author: Gobin Sougrakpam (gobins@github)
+version_added: "2.5"
+short_description: Manage AWS Direct Connect Gateway.
+description:
+  - Creates AWS Direct Connect Gateway
+  - Deletes AWS Direct Connect Gateway
+  - Attaches Virtual Gateways to Direct Connect Gateway
+  - Detaches Virtual Gateways to Direct Connect Gateway
+requirements: [ boto3 ]
+options:
+  state:
+    description:
+        - present to ensure resource is created.
+        - absent to remove resource
+    required: false
+    default: present
+    choices: [ "present", "absent"]
+  name:
+    description:
+        - name of the dxgw to be created or deleted
+    required: false
+  amazon_asn:
+    description:
+        - amazon side asn
+    required: true
+  direct_connect_gateway_id:
+    description:
+        - id of an existing direct connect gateway
+    required: false
+  virtual_gateway_id:
+    description:
+        - vpn gateway id of an existing virtual gateway
+    required: false
+'''
+
+EXAMPLES = '''
+- name: Create a new direct connect gateway attached to virtual private gateway
+  dxgw:
+    state: present
+    name: my-dx-gateway
+    amazon_asn: 7224
+    virtual_gateway_id: vpg-12345
+  register: created_dxgw
+
+- name: Create a new unattached dxgw
+  dxgw:
+    state: present
+    name: my-dx-gateway
+    amazon_asn: 7224
+  register: created_dxgw
+
+'''
+
+RETURN = '''
+result:
+  description:
+    - The attributes of the Direct Connect Gateway
+  type: complex
+  returned: I(state=present)
+  contains:
+    amazon_side_asn:
+      description: ASN on the amazon side.
+    direct_connect_gateway_id:
+      description: The ID of the direct connect gateway.
+    direct_connect_gateway_name:
+      description: The name of the direct connect gateway.
+    direct_connect_gateway_state:
+      description: The state of the direct connect gateway.
+    owner_account:
+      description: The AWS account ID of the owner of the direct connect gateway.
+'''
+
+import time
+import traceback
+
+try:
+    import botocore
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (camel_dict_to_snake_dict, ec2_argument_spec, HAS_BOTO3,
```

Comment:
You can either import HAS_BOTO3 or set it like you're doing on lines 93 and 95 - you don't need both since the imported HAS_BOTO3 was set in the same way.